### ISSUE TITLE
Segmentation fault when pushing twice in incremental mode

### ISSUE THREAD

**Issue by jad-hamza** [CVC4/CVC4#2568]
The issue was found here: https://github.com/epfl-lara/stainless/issues/363. It appears on CVC4 1.6 and on the latest master (b1fe934) 

```smt
(set-logic ALL)

(declare-datatypes () ((Unit (uu))))

(declare-fun y () Int)
(declare-fun b () Bool)
(declare-fun s () (Set Int))
(declare-fun u () Unit)

(assert (= s (insert y s)))
(assert (=> b (= uu u)))

(push 1)
(check-sat)
(pop 1)

(push 1)
(check-sat)
```

It might be related to #2401, though I get the same error with or without `--no-simp-ite-compress` 

```bash
$ cvc4 --incremental --lang smt2.5 cvc4-crash.smt2 
sat
CVC4 suffered a segfault.
Offending address is 0x10
Looks like a NULL pointer was dereferenced.
Aborted
```

**Comment by ajreynol** [comment]
Thanks a lot for the benchmark, this is fixed in PR #2572, which will be merged to master shortly.

**Comment by jad-hamza** [comment]
Awesome, thanks!
